fx.go

```diff
@@ -13,16 +13,46 @@ import (
 type ServiceIn struct {
 	fx.In
 
-	// Ancla Client.
-	BasicClient *chrysom.BasicClient
+	// PushReader is the user provided db client.
+	// (Optional)
+	// If provided, Ancla will not use default Argus db client.
+	PushReader chrysom.PushReader    `optional:"true"`
+	Options    chrysom.ClientOptions `group:"client_options"`
 }
 
-// ProvideService builds the Argus client service from the given configuration.
-func ProvideService(in ServiceIn) Service {
-	return NewService(in.BasicClient)
+type ProvideServiceOut struct {
+	fx.Out
+
+	// Ancla service.
+	Service Service
+	// Ancla listener's db client.
+	Reader chrysom.Reader
 }
 
-// TODO: Refactor and move Watch and Listener related code to chrysom.
+// ProvideService provides the Argus client service from the given configuration and client options.
+func ProvideService(in ServiceIn) (ProvideServiceOut, error) {
+	// If the user provides a non-nil chrysom.PushReader (their own db client), then use that instead
+	// of an Argus db client.
+	// Otherwise, create and use a new Argus db client.
```

Comment:
moved the logic from `ProvideBasicClient` into here, allowing users to define their own `PushReader` client instead of using the default argus db client.